### ISSUE TITLE
update coding standard to note that public documentation should be preferred

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6189]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#6189|
|Type|enhancement|
|Created|2012-11-16 01:09:02Z|
|Branch|https://github.com/twisted/twisted/tree/init-attr-doc-guide-6189|

See this message for more detail:

http://article.gmane.org/gmane.comp.python.twisted/25015

<details><summary>Searchable metadata</summary>

```
trac-id__6189 6189
type__enhancement enhancement
reporter__glyph glyph
priority__highest highest
milestone__ 
branch__branches_init_attr_doc_guide_6189 branches/init-attr-doc-guide-6189
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1353028142000000 1353028142000000
changetime__1354633126000000 1354633126000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [36422]) Branching to 'init-attr-doc-guide-6189'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/init-attr-doc-guide-6189)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_3"></a>|@therve set owner to @exarkun|
|-|-|

That looks OK. I'm still a bit confused though: it seems to me that documenting @param is not enough to know if it's going to be available as an instance attribute? I'm mostly thinking about the case where you have a "foo" parameter that ends up in "self.foo". It should probably be stated explicitly in the @param text when it's the case. Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun removed owner|
|-|-|

Hmmm.  I tried something.  I wonder if anyone thinks it's any good.  [Here's the diff](http://twistedmatrix.com/~diffresource.twistd/6189)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve set owner to @exarkun|
|-|-|

Looks great, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [36501]) Merge init-attr-doc-guide-6189 

Author: exarkun
Reviewer: therve
Fixes: [#6189](https://github.com/twisted/twisted/issues/6189)

Lay out, along with an example, the preferred approach for documenting initializer
arguments which do nothing more than specify the initial value for an attribute
sharing (or mostly sharing) their name.
